### PR TITLE
docs(aiplatform): Add GeminiQuickstart.cs #2460

### PR DESCRIPTION
**Pull request by meteatamel**

### CODE REVIEW

File: aiplatform/api/AIPlatform.Samples.Tests/GeminiQuickstartTest.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 Google LLC
```
Comment (amanda-tarafa) [inline aiplatform/api/AIPlatform.Samples.Tests/GeminiQuickstartTest.cs:2]: nit: 2024

File: aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                FileData = new() {
+                    MimeType = "image/png",
+                    FileUri = imageUri
+                }
+            }
+        });
+
+        var generateContentRequest = new GenerateContentRequest
+        {
+            Model = $"projects/{projectId}/locations/{location}/publishers/{publisher}/models/{model}",
```
Comment (amanda-tarafa) [inline aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs:64]: I'm guessing we don't have a resource name helper for this? Something like ModelName.FromProjectLocationPublisher...?

File: aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        generateContentRequest.Contents.Add(content);
+
+        // Make the request, returning a streaming response
+        using PredictionServiceClient.StreamGenerateContentStream response = predictionServiceClient.StreamGenerateContent(generateContentRequest);
+
+        StringBuilder fullText = new();
+
+        // Read streaming responses from server until complete
+        AsyncResponseStream<GenerateContentResponse> responseStream = response.GetResponseStream();
+        while (await responseStream.MoveNextAsync())
```
Comment (amanda-tarafa) [inline aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs:82]: `await foreach` maybe? See https://cloud.google.com/dotnet/docs/reference/Google.Api.Gax/latest/Google.Api.Gax.Grpc.AsyncResponseStream-1#remarks

**Review by amanda-tarafa — COMMENTED** [review]
Looks good, just a couple of comments for consideration, and a tiny nit.

File: aiplatform/api/AIPlatform.Samples.Tests/GeminiQuickstartTest.cs (hunk 1 above)
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples.Tests/GeminiQuickstartTest.cs]: Fixed

File: aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs (hunk 3 above)
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs]: Good call, fixed

File: aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs (hunk 2 above)
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/GeminiQuickstart.cs:64]: Nope, no such resource helper. There should be one, maybe we can open a bug?

**Comment by amanda-tarafa** [comment]
> @amanda-tarafa Any idea where I'd open a bug for [Model resource](https://github.com/googleapis/googleapis/blob/7d73737e0e826ac683a4b5b19352feabf51bbc57/google/cloud/aiplatform/v1/model.proto#L38-L42)? The GitHub page doesn't have an issue tracker

Just to close the loop, I've just replied internally.

**Pull request merged**